modules/grpc/otel/otel-dest-worker.cpp

```diff
@@ -56,16 +76,325 @@ DestWorker::disconnect()
+      resource_metrics->set_schema_url(current_msg_metadata.resource_schema_url);
+    }
+
+  ScopeMetrics *scope_metrics = nullptr;
+  for (int i = 0; i < resource_metrics->scope_metrics_size(); i++)
+    {
+      ScopeMetrics *possible_scope_metrics = resource_metrics->mutable_scope_metrics(i);
+      if (MessageDifferencer::Equals(possible_scope_metrics->scope(), current_msg_metadata.scope) &&
+          possible_scope_metrics->schema_url() == current_msg_metadata.scope_schema_url)
+        {
+          scope_metrics = possible_scope_metrics;
+          break;
+        }
+    }
+  if (!scope_metrics)
+    {
+      scope_metrics = resource_metrics->add_scope_metrics();
+      scope_metrics->mutable_scope()->CopyFrom(current_msg_metadata.scope);
+      scope_metrics->set_schema_url(current_msg_metadata.scope_schema_url);
+    }
+
+  return scope_metrics;
+}
+
+ScopeSpans *
+DestWorker::lookup_scope_spans(LogMessage *msg)
+{
+  if (!get_metadata_for_current_msg(msg))
+    return nullptr;
+
+  ResourceSpans *resource_spans = nullptr;
+  for (int i = 0; i < trace_service_request.resource_spans_size(); i++)
+    {
+      ResourceSpans *possible_resource_spans = trace_service_request.mutable_resource_spans(i);
+      if (MessageDifferencer::Equals(possible_resource_spans->resource(), current_msg_metadata.resource) &&
+          possible_resource_spans->schema_url() == current_msg_metadata.resource_schema_url)
+        {
+          resource_spans = possible_resource_spans;
+          break;
+        }
+    }
+  if (!resource_spans)
+    {
+      resource_spans = trace_service_request.add_resource_spans();
+      resource_spans->mutable_resource()->CopyFrom(current_msg_metadata.resource);
+      resource_spans->set_schema_url(current_msg_metadata.resource_schema_url);
+    }
+
+  ScopeSpans *scope_spans = nullptr;
+  for (int i = 0; i < resource_spans->scope_spans_size(); i++)
+    {
+      ScopeSpans *possible_scope_spans = resource_spans->mutable_scope_spans(i);
+      if (MessageDifferencer::Equals(possible_scope_spans->scope(), current_msg_metadata.scope) &&
+          possible_scope_spans->schema_url() == current_msg_metadata.scope_schema_url)
+        {
+          scope_spans = possible_scope_spans;
+          break;
+        }
+    }
+  if (!scope_spans)
+    {
+      scope_spans = resource_spans->add_scope_spans();
+      scope_spans->mutable_scope()->CopyFrom(current_msg_metadata.scope);
+      scope_spans->set_schema_url(current_msg_metadata.scope_schema_url);
+    }
+
+  return scope_spans;
+}
+
+bool
+DestWorker::insert_log_record_from_log_msg(LogMessage *msg)
+{
+  ScopeLogs *scope_logs = lookup_scope_logs(msg);
+  if (!scope_logs)
+    return false;
+
+  LogRecord *log_record = scope_logs->add_log_records();
+  return formatter.format(msg, *log_record);
+}
+
+bool
+DestWorker::insert_fallback_log_record_from_log_msg(LogMessage *msg)
+{
+  ScopeLogs *scope_logs = lookup_scope_logs(msg);
+  if (!scope_logs)
+    return false;
```

Comment:
The fallback should not fail.

I believe ` lookup_scope_logs()`, and especially `get_metadata_for_current_msg()` should be implemented differently for the fallback.

Reply:
As we talked IRL, failing to get the metadata will result in sending the message without metadata in all cases (log, fallback log, metrics, traces), instead of dropping.